### PR TITLE
Plasma golem tweaks

### PR DESCRIPTION
**Pull request by XDTM**
:cl: optional name here
fix: Plasma Golems now have the correct updated tip.
tweak: Plasma Golems no longer take damage from fire, and take normal instead of increased burn damage from other sources. They will still explode if on fire and hot enough.
add: Plasma Golems can now self-ignite with an action button.
tweak: Plasma Golems are now warned when they're close to exploding.
tweak: Plasma Golems now have a constant chance of exploding after 850K instead of a precise threshold at 900K, making it less predictable.
/:cl:

[why]: # (Please add a short description [on the next line] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding:)
# Why
Plasma Golems are kind of shunned, because their role is pretty much just a downside. Now it still is geared toward killing themselves spectacularly, but at least trying to do so doesn't require special equipment and doesn't require them to get hurt to crit first.

I'm considering increasing the explosion size a bit, given that they do kill themselves for it and are very visible when doing so. Also maybe increasing their speed while on fire.

### CODE REVIEW

**Comment by Armhulen** [comment]
Why are plasma golems these crazy suicide bombs, though? Golem ship doesn't need them, they're peaceful. Normal xenobiologists don't need them because even in the case of self defense bombing the station is a great way to get bwoinked, and finally, traitorous xenobiologists could use them but armies of punchgolems will always be stronger, and plasma for xenobiologists is rare enough for this to be a really hard strat to pull off.

**Comment by Armhulen** [comment]
And most of all: Plasma is a crazy magicfuel. why do we have to go the boring route with it when it has this much potential?

**Comment by XDTM** [comment]
If skeletons made of plasma are extremely flammable i'd expect a hulking mass of plasma to cause some discomfort around it when it destabilizes. Also we still have more materials to make more golem niches; and someone wanting to wage war through golems could desire one that can run through the enemy ranks spreading fire before blowing up, even if peacegolems might not.

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 1]
```diff
@@ -60,28 +60,62 @@
...
+	name = "Ignite"
+	desc = "Set yourself aflame, bringing yourself closer to exploding!"
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "sacredflame"
+
+/datum/action/innate/ignite/Activate()
+	to_chat(owner, "<span class='notice'>You ignite yourself!</span>")
+	if(ishuman(owner))
+		var/mob/living/carbon/human/H = owner
+		H.IgniteMob() //firestacks are already there passively
```
Comment (swindly) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: A golem can lose its firestacks and activate this before the next life tick.

Comment (XDTM) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: Yes, or in space, in which case it shouldn't ignite. I guess i could suppress the message if it doesn't ignite, but i didn't think it was too important.

**Comment by Repukan** [comment]
Mind nerfing the stupid OP sand ones?

**Comment by FantasticFwoosh** [comment]
"sand ones" die super easily to fire, they wont last two seconds in a storm.

**Comment by Repukan** [comment]
Yeah but they can absolutely butt fuck colossus and are a guaranteed VoG and crystal

**Comment by XDTM** [comment]
How about making colossus bolts half brute half burn

**Comment by Armhulen** [comment]
That would work

**Comment by ShadowDeath6** [comment]
optional name here

**Comment by lzimann** [comment]
didn't even know we had plasma golems

**Pull request merged**